onnxruntime/core/common/cpuid_info.cc

```diff
@@ -42,7 +45,15 @@ static inline int XGETBV() {
 }
 #endif  // PLATFORM_X86
 
-CPUIDInfo::CPUIDInfo() noexcept {
+CPUIDInfo CPUIDInfo::instance_;
+
+
+common::Status CPUIDInfo::Init() {
+  if (!cpuinfo_initialize()) {
```

Comment:
could we just throw from the constructor if unsuccessful?
maybe it would be simpler that way without requiring an additional call to Init().

Reply:
In a global variable like this, the constructor is called at a time we can not control. Adding lots of code in the constructor always worries me. It may introduce unnecessary complications or dependencies during startup.